From f23ccf473ea990ddba3d48730b68f23efc1f712c Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 24 Apr 2015 09:45:41 -0400
Subject: [PATCH] RHEL 7: Fix list of supported sound cards to match RHEL qemu
 (RHBZ#1176493).

---
 v2v/utils.ml | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/v2v/utils.ml b/v2v/utils.ml
index 26a3976..477033d 100644
--- a/v2v/utils.ml
+++ b/v2v/utils.ml
@@ -75,13 +75,14 @@ let kvm_arch = function
 (* Does qemu support the given sound card? *)
 let qemu_supports_sound_card = function
   | AC97
-  | ES1370
   | ICH6
   | ICH9
   | PCSpeaker
+    -> true
+  | ES1370
   | SB16
   | USBAudio
-    -> true
+    -> false
 
 let compare_app2_versions app1 app2 =
   let i = compare app1.Guestfs.app2_epoch app2.Guestfs.app2_epoch in
-- 
1.8.3.1

