From a44d5274ff35d33aea74f49d80c02f2202c766eb Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 30 Apr 2015 17:20:26 -0400
Subject: [PATCH] RHEL 7: v2v: efi: Remove references to Fedora / kraxel's OVMF
 packages.

---
 v2v/utils.ml | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/v2v/utils.ml b/v2v/utils.ml
index e9d3262..ebf799f 100644
--- a/v2v/utils.ml
+++ b/v2v/utils.ml
@@ -88,25 +88,18 @@ let qemu_supports_sound_card = function
 let find_uefi_firmware guest_arch =
   let files =
     match guest_arch with
-    | "i386" | "i486" | "i586" | "i686" ->
-       [ "/usr/share/edk2.git/ovmf-ia32/OVMF_CODE-pure-efi.fd",
-         "/usr/share/edk2.git/ovmf-ia32/OVMF_VARS-pure-efi.fd" ]
     | "x86_64" ->
        [ "/usr/share/OVMF/OVMF_CODE.fd",
-         "/usr/share/OVMF/OVMF_VARS.fd";
-         "/usr/share/edk2.git/ovmf-x64/OVMF_CODE-pure-efi.fd",
-         "/usr/share/edk2.git/ovmf-x64/OVMF_VARS-pure-efi.fd" ]
+         "/usr/share/OVMF/OVMF_VARS.fd" ]
     | "aarch64" ->
        [ "/usr/share/AAVMF/AAVMF_CODE.fd",
-         "/usr/share/AAVMF/AAVMF_VARS.fd";
-         "/usr/share/edk2.git/aarch64/QEMU_EFI-pflash.raw",
-         "/usr/share/edk2.git/aarch64/vars-template-pflash.raw" ]
+         "/usr/share/AAVMF/AAVMF_VARS.fd" ]
     | arch ->
        error (f_"don't know how to convert UEFI guests for architecture %s")
              guest_arch in
   let rec loop = function
     | [] ->
-       error (f_"cannot find firmware for UEFI guests.\n\nYou probably need to install OVMF, or Gerd's firmware repo (https://www.kraxel.org/repos/), or AAVMF (if using aarch64)")
+       error (f_"cannot find firmware for UEFI guests.\n\nYou probably need to install OVMF, or AAVMF (if using aarch64)")
     | ((code, vars_template) as ret) :: rest ->
        if Sys.file_exists code && Sys.file_exists vars_template then ret
        else loop rest
-- 
1.8.3.1

